metas.groovy

```diff
@@ -0,0 +1,137 @@
+//Adapted from module.groovy
+@GrabResolver(name = 'jcenter', root = 'http://jcenter.bintray.com/')
+@Grab(group='org.ajoberstar', module='grgit', version='1.9.3')
+import org.ajoberstar.grgit.Grgit
+import groovy.json.JsonSlurper
+
+Properties properties = new Properties()
+new File("gradle.properties").withInputStream {
+    properties.load(it)
+}
+
+githubHome = properties.alternativeGithubHome ?: "MetaTerasology"
+
+modulesRetrieved = []
+
+
+def retrieve(String[] modules) {
+    println "Now inside retrieve, user wants: $modules"
+    for (String module : modules) {
+        println "Starting loop for meta module $module"
+        println "Meta Modules retrieved so far: $modulesRetrieved"
+        retrieveModule(module)
+    }
+}
+
+def retrieveModule(String module) {
+    File targetDir = new File("meta/$module")
+    println "Request to retrieve meta module $module would store it at $targetDir - exists? " + targetDir.exists()
+
+    if (targetDir.exists()) {
+        println "That meta module already had an existing directory locally. If something is wrong with it please delete and try again"
+        modulesRetrieved << module
+    } else if (modulesRetrieved.contains(module)) {
+        println "We already retrieved $module - skipping"
+    } else {
+        println "Retrieving meta module $module - if it doesn't appear to exist (typo for instance) you'll get an auth prompt (in case it is private)"
+        //noinspection GroovyAssignabilityCheck - GrGit has its own .clone but a warning gets issued for Object.clone
+        Grgit.clone dir: targetDir, uri: "https://github.com/$githubHome/${module}.git"
+        modulesRetrieved << module
+
+        // TODO: Temporary until build.gradle gets removed from module directories (pending Cervator work from module.groovy)
+        // File targetBuildGradle = new File(targetDir, 'build.gradle')
+        // targetBuildGradle.delete()
+        // targetBuildGradle << new File('templates/build.gradle').text
+
+    }
+}
+
+
+def createMetaModule(String name) {
+    // Check if the module already exists. If not, create the module directory
+    File targetDir = new File("meta/$name")
+    if (targetDir.exists()) {
+        println "Target directory already exists. Aborting."
+        return
+    }
+    println "Creating target directory"
+    targetDir.mkdir()
+
+    // Add gitignore
+    println "Creating .gitignore"
+    File gitignore = new File(targetDir, ".gitignore")
+    def gitignoreText = new File("templates/.gitignore").text
+    gitignore << gitignoreText
+
+    // Initialize git
+    Grgit.init dir: targetDir, bare: false
+}
+
+def getUserString (String prompt) {

```

Comment:
I'm thinking we could put the shared methods somewhere to only have them in one place and just reuse. Hmm. Idea forming.

Reply:
Yeah... that would be better for the other tasks....possibly a task in itself... *maybe*